nginx/templates/_gzip.jinja

```diff
@@ -0,0 +1,29 @@
+{%- if gzip is defined and gzip is mapping %}
+{% from "nginx/lib.sls" import configurable %}
+  # Start configure gzip for vhost
+  gzip {{ 'on' if gzip.enable|default(true) else 'off' }};
```

Comment:
It only applies this when the `gzip` key is present in the pillar and if it is a mapping. It will not enable gzip for every vhost we already have :)